### ISSUE TITLE
URock questions - wind speed height and roughness

### ISSUE THREAD

**Issue by VladimirSobral** [UMEP-dev/UMEP-processing#84]
Hi,

Could anyone clarify some questions about URock model?
I created some wind speed maps with the URock model with the output height of 1.50 m. 
However, the maps show wind speed information over buildings. Does this information refer to the height above the roof of the buildings? And how could I have a map without this information (just the streets)?
The other question is: if I understood correctly from the article by Bernard et al. (2023), URock calculates the average roughness value of the study area for wind mapping. Where can I find this value?

Thank you!

**Comment by j3r3m1** [comment]
Hi VladimirSobral,
Is the wind speed over building equal to 0 ?
- If not, there is probably an error with the input data you have used since the wind speed refer to the one at height above ground level. Can you share your data ?
- If yes putting the building layer above your wind speed layer in QGIS may make the trick of removing the 0 m/s wind speed within your building polygons

Concerning the roughness lengths, it should now be displayed within the QGIS log box.

**Comment by VladimirSobral** [comment]
Hi @j3r3m1 

No. The wind speed over buildings is not zero, as shown in Figures. It's interesting because I had a similar result with SOLWEIG. However, when I use SpatiaTC to map the PET index, the buildings are subtracted from the results.
Yes, I'm attaching the data. 
![output_UrockWS](https://github.com/user-attachments/assets/28dc3ef1-723b-466f-800f-028bb93c6d9e)
![output_Urock-BUILDINGS](https://github.com/user-attachments/assets/3c57e439-59fa-44c8-8fce-08f1dfae5758)
[UROCK.zip](https://github.com/user-attachments/files/16750934/UROCK.zip)

Thank you again,

**Comment by j3r3m1** [comment]
Can you share the parameters you have used for the calculation (the one that are in the history of the toolbox) ?

**Comment by VladimirSobral** [comment]
I can't find the history of the toolbox. Could you tell me where it is?
Anyway, I'm sending you some screenshots of the Urock parameters.

![1](https://github.com/user-attachments/assets/e4226c80-c64c-4852-840a-80235c16c91a)
![2](https://github.com/user-attachments/assets/eabb6ee1-cfdb-4fbd-8ea9-26c99472d574)

**Comment by VladimirSobral** [comment]
Hi @j3r3m1 

Thank you so much for your comments. I ran other simulations with the parameters you suggested.
Please let me share the results with you. In the first file, I used the vertical resolution of 3 meters. In the second file, I changed the output wind height (1.0 m). In two cases, the URock maps the wind speed over buildings.
Should I extract the building areas using some raster function?

[Urock.zip](https://github.com/user-attachments/files/16769625/Urock.zip)

**Comment by j3r3m1** [comment]
Good catch, there is indeed an issue with the interpolation. The calculations are made using a grid oriented toward the wind flow while the resulting raster is oriented toward x and y axis. Thus there is an interpolation which is performed to fill the cells of the raster and this interpolation creates non-null value for building cells. I will correct that.
For the moment, you can use the vector output which does not have this issue.

**Comment by VladimirSobral** [comment]
I got it! Thank you @j3r3m1. 
just a final question regarding roughness lengths...I can't find the zip code of the latest URock version. 
As I'm unfamiliar with codes.... could you explain how to update the URock?

**Comment by j3r3m1** [comment]
Sure ! You can download the code as zip going on the code page (https://github.com/UMEP-dev/UMEP-processing) and clicking on the green button "Code" and then "Download as zip".
Then in QGIS you can install this code using "Plugin"-> "Manage and install plugins" -> "Install from zip".

**Comment by j3r3m1** [comment]
I have updated the code. If you use the last version, you should now have exactly 0 when there is a building.

**Comment by j3r3m1** [comment]
To @VladimirSobral I close the issue but feel free to reopen if you still meet some problems.

**Comment by VladimirSobral** [comment]
Thank you @j3r3m1 

I did two tests. First, I used a vertical resolution of 10 m, which was ok. But when I used a vertical resolution of 3 m, I had this problem:

Traceback (most recent call last):
File "C:\OSGeo4W/apps/qgis-ltr/./python/plugins\processing\algs\gdal\GdalUtils.py", line 130, in on_stderr
val = ba.data().decode('UTF-8')
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xa2 in position 46: invalid start byte

The above exception was the direct cause of the following exception:

SystemError: <class 'PyQt5.QtCore.QByteArray'> returned a result with an error set

The above exception was the direct cause of the following exception:

SystemError: <class 'PyQt5.QtCore.QByteArray'> returned a result with an error set

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
File "C:\OSGeo4W/apps/qgis-ltr/./python/plugins\processing\algs\gdal\gdalcalc.py", line 198, in processAlgorithm
return GdalAlgorithm.processAlgorithm(self, parameters, context, feedback)
File "C:\OSGeo4W/apps/qgis-ltr/./python/plugins\processing\algs\gdal\GdalAlgorithm.py", line 135, in processAlgorithm
GdalUtils.runGdal(commands, feedback)
File "C:\OSGeo4W/apps/qgis-ltr/./python/plugins\processing\algs\gdal\GdalUtils.py", line 146, in runGdal
res = proc.run(feedback)
SystemError: <built-in method run of QgsBlockingProcess object at 0x0000021558E399D0> returned a result with an error set

Execution failed after 1847.15 segundos (30 minutos 47 segundos)

What could have happened?
